File: VeamIt-project/iOS/veam00000000/ConsoleUpdatePreparingSellPdfStatusHandler.m
//
//  ConsoleUpdatePreparingSellPdfStatusHandler.m
//  veam00000000
//
//  Created by veam on 11/5/15.
//  Copyright (c) 2015 veam. All rights reserved.
//

#import "ConsoleUpdatePreparingSellPdfStatusHandler.h"

#import "ConsoleUtil.h"

@implementation ConsoleUpdatePreparingSellPdfStatusHandler

- (void)handlePostResult:(NSArray *)results
{
    //NSLog(@"%@::handlePostResult",NSStringFromClass([self class])) ;
    if([results count] >= 2){
        //NSLog(@"count >= 2") ;
        NSString *result = [results objectAtIndex:0] ;
        if([result isEqualToString:@"OK"]){
            ConsoleContents *contents = [ConsoleUtil getConsoleContents] ;
            NSString *elementString = [results objectAtIndex:1] ;
            //NSLog(@"elementString=%@",elementString) ;
            NSArray *elements = [elementString componentsSeparatedByString:@","] ;
            int numberOfElements = [elements count] ;
            for(int index = 0 ; index < numberOfElements ; index++){
                NSString *valueString = [elements objectAtIndex:index] ;
                NSArray *values = [valueString componentsSeparatedByString:@"|"] ;
                if([values count] >= 3){
                    NSString *sellPdfId = [values objectAtIndex:0] ;
                    NSString *status = [values objectAtIndex:1] ;
                    NSString *statusText = [values objectAtIndex:2] ;
                    SellPdf *sellPdf = [contents getSellPdfForId:sellPdfId] ;
                    if(sellPdf != nil){
                        [sellPdf setStatus:status] ;
                        [sellPdf setStatusText:statusText] ;
                    }
                } else {
                    //NSLog(@"%@ less than 3",valueString) ;
                }
            }
        }
    }
}

@end

File: VeamIt-project/Android/veam31000287/veam31000287/src/main/java/co/veam/veam31000287/ConsoleUpdatePreparingSellPdfStatusHandler.java
package co.veam.veam31000287;

import java.util.ArrayList;

/**
 * Created by veam on 11/8/16.
 */
public class ConsoleUpdatePreparingSellPdfStatusHandler implements HandlePostResultInterface {
    @Override
    public void handlePostResult(ArrayList<String> results) {
        //NSLog(@"%@::handlePostResult",NSStringFromClass(this.class])) ;
        if(results.size()  >= 2){
            //NSLog(@"count >= 2") ;
            String result = results.get(0) ;
            if(result.equals("OK")){
                ConsoleContents contents = ConsoleUtil.getConsoleContents() ;
                String elementString = results.get(1) ;
                //NSLog(@"elementString=%@",elementString) ;
                String[] elements = elementString.split(",") ;
                int numberOfElements = elements.length ;
                for(int index = 0 ; index < numberOfElements ; index++){
                    String valueString = elements[index] ;
                    String[] values = valueString.split("\\|") ;
                    if(values.length >= 3){
                        String sellPdfId = values[0] ;
                        String status = values[1] ;
                        String statusText = values[2] ;
                        SellPdfObject sellPdf = contents.getSellPdfForId(sellPdfId) ;
                        if(sellPdf != null){
                            sellPdf.setStatus(status) ;
                            sellPdf.setStatusText(statusText) ;
                        }
                    } else {
                        //NSLog(@"%@ less than 3",valueString) ;
                    }
                }
            }
        }

    }
}
